Stabilize testPersistenceMultipleNodesClientSessionsAtRandomNode model test

### Describe the bug

testPersistenceMultipleNodesClientSessionsAtRandomNode is failing intermittently with the following stacktrace:
```
ERROR [org.infinispan.interceptors.impl.InvocationContextInterceptor] (jgroups-10,node-8) ISPN000136: Error executing command TouchCommand on Cache 'offlineSessions', writing keys []
 org.infinispan.remoting.transport.jgroups.SuspectException: ISPN000400: Node node-7 was suspected
 	at org.infinispan.remoting.transport.ResponseCollectors.remoteNodeSuspected(ResponseCollectors.java:31)
 	at org.infinispan.remoting.transport.impl.SingleResponseCollector.targetNotFound(SingleResponseCollector.java:31)
 	at org.infinispan.remoting.transport.impl.SingleResponseCollector.targetNotFound(SingleResponseCollector.java:17)
 	at org.infinispan.remoting.transport.ValidSingleResponseCollector.addResponse(ValidSingleResponseCollector.java:23)
 	at org.infinispan.remoting.transport.impl.SingleTargetRequest.addResponse(SingleTargetRequest.java:74)
 	at org.infinispan.remoting.transport.impl.SingleTargetRequest.onResponse(SingleTargetRequest.java:44)
 	at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:51)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1496)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1398)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.access$300(JGroupsTransport.java:146)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1586)
 	at org.jgroups.JChannel.up(JChannel.java:780)
 	at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:913)
 	at org.jgroups.protocols.FRAG3.up(FRAG3.java:165)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.pbcast.GMS.up(GMS.java:876)
 	at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:254)
 	at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1048)
 	at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:771)
 	at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:752)
 	at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:405)
 	at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:592)
 	at org.jgroups.protocols.VERIFY_SUSPECT.up(VERIFY_SUSPECT.java:132)
 	at org.jgroups.protocols.FailureDetection.up(FailureDetection.java:186)
 	at org.jgroups.protocols.FD_SOCK.up(FD_SOCK.java:254)
 	at org.jgroups.protocols.MERGE3.up(MERGE3.java:281)
 	at org.jgroups.protocols.Discovery.up(Discovery.java:300)
 	at org.jgroups.protocols.TP.passMessageUp(TP.java:1400)
 	at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:98)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:829)
 01:06:44,314 ERROR [org.infinispan.interceptors.impl.InvocationContextInterceptor] (jgroups-10,node-8) ISPN000136: Error executing command GetCacheEntryCommand on Cache 'offlineSessions', writing keys []
 java.lang.IllegalStateException: Read commands must ignore leavers
 	at org.infinispan.interceptors.impl.BaseStateTransferInterceptor.handleExceptionOnReadCommandReturn(BaseStateTransferInterceptor.java:211)
 	at org.infinispan.interceptors.impl.BaseStateTransferInterceptor.handleReadCommandReturn(BaseStateTransferInterceptor.java:254)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:124)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:87)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:33)
 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
 	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:114)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:87)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:33)
 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
 	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
 	at org.infinispan.remoting.transport.AbstractRequest.completeExceptionally(AbstractRequest.java:74)
 	at org.infinispan.remoting.transport.impl.SingleTargetRequest.onResponse(SingleTargetRequest.java:48)
 	at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:51)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1496)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1398)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.access$300(JGroupsTransport.java:146)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1586)
 	at org.jgroups.JChannel.up(JChannel.java:780)
 	at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:913)
 	at org.jgroups.protocols.FRAG3.up(FRAG3.java:165)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.pbcast.GMS.up(GMS.java:876)
 	at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:254)
 	at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1048)
 	at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:771)
 	at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:752)
 	at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:405)
 	at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:592)
 	at org.jgroups.protocols.VERIFY_SUSPECT.up(VERIFY_SUSPECT.java:132)
 	at org.jgroups.protocols.FailureDetection.up(FailureDetection.java:186)
 	at org.jgroups.protocols.FD_SOCK.up(FD_SOCK.java:254)
 	at org.jgroups.protocols.MERGE3.up(MERGE3.java:281)
 	at org.jgroups.protocols.Discovery.up(Discovery.java:300)
 	at org.jgroups.protocols.TP.passMessageUp(TP.java:1400)
 	at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:98)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:829)
 01:06:44,315 WARN  [org.infinispan.CLUSTER] (jgroups-10,node-8) ISPN000071: Caught exception when handling command ClusteredGetCommand{key=32f4760a-55de-4d10-873b-73ea599e4a5d, flags=[], topologyId=9, isWrite=false}
 java.lang.IllegalStateException: Read commands must ignore leavers
 	at org.infinispan.interceptors.impl.BaseStateTransferInterceptor.handleExceptionOnReadCommandReturn(BaseStateTransferInterceptor.java:211)
 	at org.infinispan.interceptors.impl.BaseStateTransferInterceptor.handleReadCommandReturn(BaseStateTransferInterceptor.java:254)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:124)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:87)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:33)
 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
 	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.invokeQueuedHandlers(QueueAsyncInvocationStage.java:114)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:87)
 	at org.infinispan.interceptors.impl.QueueAsyncInvocationStage.accept(QueueAsyncInvocationStage.java:33)
 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
 	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
 	at org.infinispan.remoting.transport.AbstractRequest.completeExceptionally(AbstractRequest.java:74)
 	at org.infinispan.remoting.transport.impl.SingleTargetRequest.onResponse(SingleTargetRequest.java:48)
 	at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:51)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1496)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1398)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.access$300(JGroupsTransport.java:146)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1586)
 	at org.jgroups.JChannel.up(JChannel.java:780)
 	at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:913)
 	at org.jgroups.protocols.FRAG3.up(FRAG3.java:165)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.pbcast.GMS.up(GMS.java:876)
 	at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:254)
 	at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1048)
 	at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:771)
 	at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:752)
 	at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:405)
 	at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:592)
 	at org.jgroups.protocols.VERIFY_SUSPECT.up(VERIFY_SUSPECT.java:132)
 	at org.jgroups.protocols.FailureDetection.up(FailureDetection.java:186)
 	at org.jgroups.protocols.FD_SOCK.up(FD_SOCK.java:254)
 	at org.jgroups.protocols.MERGE3.up(MERGE3.java:281)
 	at org.jgroups.protocols.Discovery.up(Discovery.java:300)
 	at org.jgroups.protocols.TP.passMessageUp(TP.java:1400)
 	at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:98)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:829)
 01:06:44,315 ERROR [org.infinispan.interceptors.impl.InvocationContextInterceptor] (jgroups-6,node-6) ISPN000136: Error executing command GetKeyValueCommand on Cache 'offlineSessions', writing keys []
 org.infinispan.remoting.RemoteException: ISPN000217: Received exception from node-8, see cause for remote stack trace
 	at org.infinispan.remoting.transport.ResponseCollectors.wrapRemoteException(ResponseCollectors.java:25)
 	at org.infinispan.interceptors.distribution.RemoteGetSingleKeyCollector.addResponse(RemoteGetSingleKeyCollector.java:34)
 	at org.infinispan.interceptors.distribution.RemoteGetSingleKeyCollector.addResponse(RemoteGetSingleKeyCollector.java:25)
 	at org.infinispan.remoting.transport.impl.MultiTargetRequest.onResponse(MultiTargetRequest.java:92)
 	at org.infinispan.remoting.transport.jgroups.StaggeredRequest.onResponse(StaggeredRequest.java:50)
 	at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:51)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1496)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1398)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.access$300(JGroupsTransport.java:146)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1586)
 	at org.jgroups.JChannel.up(JChannel.java:780)
 	at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:913)
 	at org.jgroups.protocols.FRAG3.up(FRAG3.java:165)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.FlowControl.up(FlowControl.java:347)
 	at org.jgroups.protocols.pbcast.GMS.up(GMS.java:876)
 	at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:254)
 	at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1048)
 	at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:771)
 	at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:752)
 	at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:405)
 	at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:592)
 	at org.jgroups.protocols.VERIFY_SUSPECT.up(VERIFY_SUSPECT.java:132)
 	at org.jgroups.protocols.FailureDetection.up(FailureDetection.java:186)
 	at org.jgroups.protocols.FD_SOCK.up(FD_SOCK.java:254)
 	at org.jgroups.protocols.MERGE3.up(MERGE3.java:281)
 	at org.jgroups.protocols.Discovery.up(Discovery.java:300)
 	at org.jgroups.protocols.TP.passMessageUp(TP.java:1400)
 	at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:98)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:829)
 Caused by: java.lang.IllegalStateException: Read commands must ignore leavers
 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
 	at org.infinispan.marshall.exts.ThrowableExternalizer.readGenericThrowable(ThrowableExternalizer.java:282)
 	at org.infinispan.marshall.exts.ThrowableExternalizer.readObject(ThrowableExternalizer.java:259)
 	at org.infinispan.marshall.exts.ThrowableExternalizer.readObject(ThrowableExternalizer.java:42)
 	at org.infinispan.marshall.core.GlobalMarshaller.readWithExternalizer(GlobalMarshaller.java:727)
 	at org.infinispan.marshall.core.GlobalMarshaller.readNonNullableObject(GlobalMarshaller.java:708)
 	at org.infinispan.marshall.core.GlobalMarshaller.readNullableObject(GlobalMarshaller.java:357)
 	at org.infinispan.marshall.core.BytesObjectInput.readObject(BytesObjectInput.java:32)
 	at org.infinispan.remoting.responses.ExceptionResponse$Externalizer.readObject(ExceptionResponse.java:49)
 	at org.infinispan.remoting.responses.ExceptionResponse$Externalizer.readObject(ExceptionResponse.java:41)
 	at org.infinispan.marshall.core.GlobalMarshaller.readWithExternalizer(GlobalMarshaller.java:727)
 	at org.infinispan.marshall.core.GlobalMarshaller.readNonNullableObject(GlobalMarshaller.java:708)
 	at org.infinispan.marshall.core.GlobalMarshaller.readNullableObject(GlobalMarshaller.java:357)
 	at org.infinispan.marshall.core.GlobalMarshaller.objectFromObjectInput(GlobalMarshaller.java:191)
 	at org.infinispan.marshall.core.GlobalMarshaller.objectFromByteBuffer(GlobalMarshaller.java:220)
 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1488)
```

### Version

main

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

--- a/testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java
+++ b/testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java
@@ -257,8 +257,9 @@ public class OfflineSessionPersistenceTest extends KeycloakModelTest {
                     // IllegalLifecycleStateException: ISPN000324: Cache 'clientSessions' is in 'STOPPING' state and this is an invocation not belonging to an
                     // on-going transaction, so it does not accept new invocations."
                     // also: org.infinispan.commons.CacheException: java.lang.IllegalStateException: Read commands must ignore leavers
-                    if ((ex.getCause() != null && ex.getCause().getMessage().contains("ISPN000324") ||
-                            (ex instanceof CacheException && ex.getMessage().contains("Read commands must ignore leavers")))) {
+                    if ((ex.getCause() != null && ex.getCause().getMessage().contains("ISPN000324")) || 
+                            (ex.getMessage() != null && ex.getMessage().contains("ISPN000217")) ||
+                            (ex instanceof CacheException && ex.getMessage().contains("Read commands must ignore leavers"))) {
                         log.warn("invocation failed, skipping. Retrying might lead to a 'Unique index or primary key violation' when the offline session has already been stored in the DB in the current session", ex);
                     } else {
                         throw ex;

